internal/layer2/template.go

```diff
@@ -0,0 +1,136 @@
+package layer2
+
+// TODO(Rahul): move this to config map
+const configTemplate = `
+package_update: true
+package_upgrade: true
+packages:
+  - jq
+  - vlan
+
+write_files:
+  - path: /var/lib/capi_network_settings/final_configuration.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+      set -euo pipefail
+
+      echo "Running final configuration commands"
+      apt-get update -qq
+      apt-get install -y -qq jq vlan
+
+      modprobe 8021q
+      echo "8021q" >> /etc/modules
+
+      # Generate the network configuration and append it to /etc/network/interfaces for each VLAN-tagged sub-interface.
```

Comment:
also add route configuration in the user-data script

Reply:
done, PTAL!